src/common/page-queries/microblock-hash.ts

```diff
@@ -0,0 +1,67 @@
+import { NextPageContext } from 'next';
+import { queryWith0x } from '@common/utils';
+import { GetQueries, getSingleCachedQueryData, Queries } from 'jotai-query-toolkit/nextjs';
+import { Microblock } from '@stacks/stacks-blockchain-api-types';
+import { getApiClients } from '@common/api/client';
+import { getTxQueryKey } from '@store/transactions';
+import { getMicroblocksQueryKey } from '@store/microblocks';
+import { QueryClient } from 'react-query';
+import { getBlocksQueryKey } from '@store/blocks';
+
+export function getMicroblockHashFromCtx(ctx: NextPageContext) {
+  const { query } = ctx;
+  if (!query?.hash) throw Error('No microblock hash');
+  return queryWith0x(query?.hash?.toString());
+}
+
+function getMicroblockPageCachedQueryProps(ctx: NextPageContext, queryClient: QueryClient) {
+  const microblockHash = getMicroblockHashFromCtx(ctx);
+  const queryKey = getMicroblocksQueryKey.single(microblockHash);
+  return getSingleCachedQueryData<Microblock>(queryKey, queryClient);
+}
+
+export const getMicroblockPageQueryProps = async (
+  ctx: NextPageContext,
+  queryClient: QueryClient
+  // TODO: Remove 'any' before deploy when types are updated in client package
```

Comment:
@fbwoolf Shall this be updated ?

Reply:
@kyranjamie do you know why when I remove `any` I get this type error here?

![image](https://user-images.githubusercontent.com/6493321/129907353-076c3be8-50a8-4739-9fa3-7d8a6a72ffed.png)

Reply:
you'll need to cast it -- the types are different between the client and api-types libraries